### PR TITLE
Little endian updates, automatic format detection

### LINKED ISSUE #15: mcpe 0.16 support 'varint'
> \<gurun\> talked to mojang today. The NBT for items, chunks and other stuff is a mess right now, even for them. They ARE moving towards all of it going over the wire to be VarInt style, but right now .. it's totally a mix. ​

I present you, the new specification!
`varint` is **always** little-endian (finally they can settle on something)
we also need `varlong`, I don't think ProtoDef has that

Here is an implementation in C# https://github.com/NiclasOlofsson/fNbt/commit/61619ac593fe78c992a292850b40b44cc7d43ccc
Strings are `varint` prefixed, and there can be `varint32` and `varint64` fields it looks like
- All ints are now replaced with `varints`
- We will need to actually separate the implementations

### PR DESCRIPTION
**Pull request by extremeheat**
Fixes #15 

* fix reading empty nbt compounds
* fix incorrect floating point reading in little endian
* add little endian + varint support for bedrock with zigzag encoding
* new promise-based API
* automatic format detection: try big, little, little-varint in succession
* sanity checks to avoid node.js crashing and allow format detection
* add some new samples

WIP. Requires https://github.com/ProtoDef-io/node-protodef/pull/112 and https://github.com/ProtoDef-io/node-protodef/pull/113 also the following:

TODO:
 - [x] add support for writing BigInts in protodef
 - [x] add support for reading at custom buffer offset in protodef
 - [ ] more reading/writing tests for little endian

### CODE REVIEW

**Comment by rom1504** [comment]
tests are not passing, you can change the package.json version of deps temporarily so CI run

**Comment by rom1504** [comment]
>  add support for writing BigInts in protodef

wouldn't it be better to simply add a new type instead of changing the existing i64 ?

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "fix": "standard --fix"
   },
   "dependencies": {
-    "protodef": "^1.7.0"
+    "protodef": "ProtoDef-io/node-protodef"
```
Comment (rom1504) [inline package.json:35]: You can use 1.9.0 now

**Comment by rom1504** [comment]
is this ready to review ?

**Comment by extremeheat** [comment]
Should be ready for review now. I didn't add any wrapper here for the bigint, so it just returns that directly.

**Review by rom1504 — APPROVED** [review]

**Pull request merged**